packages/react-instantsearch-core/src/core/indexUtils.js

```diff
@@ -161,44 +167,99 @@ function getNamespaceAndAttributeName(id) {
   return { namespace, attributeName };
 }
 
-// eslint-disable-next-line max-params
+function hasRefinements({
+  context,
+  indexId,
+  namespace,
+  attributeName,
+  id,
+  searchState,
+}) {
+  const multiIndex = hasMultipleIndices(context);
+
+  if (multiIndex && namespace) {
+    return (
+      searchState &&
+      searchState.indices &&
+      searchState.indices[indexId] &&
+      searchState.indices[indexId][namespace] &&
+      searchState.indices[indexId][namespace].hasOwnProperty(attributeName)
+    );
+  }
+
+  if (multiIndex) {
+    return (
+      searchState &&
+      searchState.indices &&
+      searchState.indices[indexId] &&
+      searchState.indices[indexId].hasOwnProperty(id)
+    );
+  }
+
+  if (namespace) {
+    return (
+      searchState &&
+      searchState[namespace] &&
+      searchState[namespace].hasOwnProperty(attributeName)
+    );
+  }
+
+  return searchState && searchState.hasOwnProperty(id);
+}
+
+function getRefinements({
+  context,
+  namespace,
+  searchState,
+  indexId,
+  attributeName,
+  id,
+}) {
+  const multiIndex = hasMultipleIndices(context);
+
+  if (multiIndex && namespace) {
+    return (
+      searchState.indices[indexId][namespace] &&
```

Comment:
We already perform those checks inside the `hasRefinements` function. It should be safe to directly access the value. The comment applies to the rest of the function.